treatment-service/models/storage.go

```diff
@@ -78,6 +78,7 @@ type ExperimentIndex struct {
 	stringSets map[string]*set.Set
 	intSets    map[string]*set.Set
 	realSets   map[string]*set.Set
+	boolFlags  map[string]bool
```

Comment:
Added this new field to the `ExperimentIndex` struct to store boolean segmenter values.

Reply:
This needs to be a `set.Set` as well. It's possible to have a bool segment as `{true, false}` if it applies to both values.

Reply:
:O Really? But the original implementation of `matchFlagSetSegment` seems to suggest that the flag could only be one of `true` or `false`

Reply:
How? We are iterating over `range i.Experiment.Segments[segmentName].GetValues()`, no?

Reply:
Ohhh I see it! Oops my bad, I got carried away with the slightly different implementation of this function as compared to the other `match___SetSegment` functions and interpreted this differently 😅